services/backend/src/core/search/search.js

```diff
@@ -72,6 +74,14 @@ async function filterSearch(request, response) {
       results = await utils.classificationSearch(results, classifications);
     }
 
+    if (!anyMentorSkills && mentorSkills) {
+      results = await utils.mentorSkillsSearch(results, mentorSkills);
+    }
+
+    if (anyMentorSkills) {
+      results = await utils.anyMentorSkillsSearch(results);
+    }
```

Comment:
Should treat anyMentorSkills like a string
```suggestion
    if (anyMentorSkills === "true") {
      results = await utils.anyMentorSkillsSearch(results);
    } else if (mentorSkills) {
      results = await utils.mentorSkillsSearch(results, mentorSkills);
    }
```